konnectivity-client/pkg/client/client_test.go

```diff
@@ -51,14 +51,64 @@ func TestDial(t *testing.T) {
 	}
 
 	if ts.packets[0].Type != client.PacketType_DIAL_REQ {
-		t.Fatalf("expect packet.type %v; got %v", client.PacketType_CLOSE_REQ, ts.packets[0].Type)
+		t.Fatalf("expect packet.type %v; got %v", client.PacketType_DIAL_REQ, ts.packets[0].Type)
 	}
 
 	if ts.packets[0].GetDialRequest().Address != "127.0.0.1:80" {
 		t.Errorf("expect packet.address %v; got %v", "127.0.0.1:80", ts.packets[0].GetDialRequest().Address)
 	}
 }
 
+// TestDialRace exercises the scenario where serve() observes and handles DIAL_RSP
+// before DialContext() does any work after sending the DIAL_REQ, and ensures
+// the handoff of the established connection doesn't error and doesn't time out.
```

Comment:
I don't think that last part is super clear. What if we don't say that and just say we had bugs if this happens in the past?

```suggestion
// before DialContext() does any work after sending the DIAL_REQ. We had bugs
// handling this case in the past, make sure we don't regress.
```